### ISSUE TITLE
have jbang-catalog based init templates 

### ISSUE THREAD

**Issue by maxandersen** [jbangdev/jbang#303]
currently we have a very crude "init template" system in jbang where the parameter passed to `--template` in init is looked up inside jbang.jar as `<template>-init.java.qute`

This is nice and simple but not extendable and not community extendable at all.

Was wondering if we should allow for using implicit alias catalog syntax here making it so you can do things like:

```
jbang init -t cli cli.java                ## does what it does today (looks locally)
jbang init -t cli@.           cli.java    ## uses init ref in github.com/jbangdev/jbang-catalog
jbang init -t rest@quarkusio  restex.java ## uses rest ref from github.com/quarkusio/jbang-catalog
etc.
```

This would allow for growing the list of templates in a centralized location while still allow for decentralized experimentatoin/specialization.

**Comment by maxandersen** [comment]
wether the templates are in jbang-catalog.json or separate in lets say jbang-init-catalog.json is to be figured out.

**Comment by agoncal** [comment]
I don't know if it's related, but I miss having a catalog of templates. In fact, I would love to have

```
$ jbang template 

Commands:
  add     Add a new template.
  list    Show currently defined templates.
  remove  Remove existing template.
```

In fact, a bit like aliases where you can create an alias, give it a name, description, etc.

**Comment by quintesse** [comment]
@agoncal that would certainly be my first idea as well: to implement it just like aliases are implemented now.
(In fact at some point we were thinking if we could just treat templates as a special kind of alias, but that idea was discarded as not being practical)

**Comment by maxandersen** [comment]
I think doing jbang-templates.json mimicking catalog as a start is good. Change it to have a map of template files  rather than a single script. 

Then treat it as qute templates. 

Key would be relative path and value the template name.

**Comment by quintesse** [comment]
I was actually thinking of having a `templates` section in the `jbang-catalog.json`. Isn't that an easier option or does that have disadvantages?

**Comment by maxandersen** [comment]
not sure - my thinking is just that the size of templates could be much larger than the jbang-catalog script...and I already fear we are being not eager enough in fetching update scripts adding templates would make the wait longer...

like, for quarkus and hibernate I could think of having tons of templates where as there would just be a handful or few scripts - thus carrying all templates around *always* seems wrong.

**Comment by quintesse** [comment]
Ok, locally I don't see much of a problem having different files, but for the remote ones I see some problems. For example we would need a separate catalog update system just for templates. Because `jbang catalog add quarkusio` would retrieve the jbang-catalog.json file, but how would we get the jbang-templates.json file? We'd need another set of `jbang template-catalog add/list/remove` commands, which doesn't seem very nice.

Is it really such a problem to have to download somewhat large .json files? I mean, most JPGs on a web page would probably already be larger.

**Comment by maxandersen** [comment]
You are right. 

Keep it simple first by using existing infrastructure. 

Let's start there and let's just measure it and see how bad it goes :)

**Comment by maxandersen** [comment]
fixed in #748
